pubsub/test/subscriptions.test.js

```diff
@@ -73,110 +74,126 @@ describe('pubsub:subscriptions', function () {
-      assert(callback.firstCall.args[0].message = '"topicName" is required!');
-      assert.equal(callback.firstCall.args[1], undefined);
+      assert(callback.firstCall.args[0].message === 'error');
     });
-    it('should require subscriptionName', function () {
+  });
+
+  describe('deleteSubscription', function () {
+    it('should delete a subscription', function () {
       var sample = getSample();
       var callback = sinon.stub();
 
-      sample.program.createSubscription(topicName, undefined, callback);
+      sample.program.deleteSubscription(subscriptionName, callback);
 
-      assert(callback.firstCall.args[0]);
-      assert(callback.firstCall.args[0].message = '"subscriptionName" is required!');
-      assert.equal(callback.firstCall.args[1], undefined);
+      assert.ifError(callback.firstCall.args[0]);
+      assert.strictEqual(callback.firstCall.args[1], sample.mocks.apiResponse);
+      assert(console.log.calledWith('Deleted subscription: %s', subscriptionName));
     });
+
     it('should handle error', function () {
       var sample = getSample();
       var error = new Error('error');
       var callback = sinon.stub();
-      sample.mocks.pubsub.subscribe.callsArgWith(3, error);
+      sample.mocks.subscription.delete.callsArgWith(0, error);
 
-      sample.program.createSubscription(topicName, subscriptionName, callback);
+      sample.program.deleteSubscription(subscriptionName, callback);
 
       assert(callback.firstCall.args[0]);
       assert(callback.firstCall.args[0].message === 'error');
     });
   });
 
-  describe('delete', function () {
-    it('should delete a subscription', function () {
+  describe('getSubscriptionMetadata', function () {
+    it('should get the metadata for a subscription', function () {
       var sample = getSample();
       var callback = sinon.stub();
 
-      sample.program.deleteSubscription(subscriptionName, callback);
+      sample.program.getSubscriptionMetadata(subscriptionName, callback);
 
       assert.ifError(callback.firstCall.args[0]);
-      assert(console.log.calledWith('Deleted subscription: %s', subscriptionName));
+      assert.strictEqual(callback.firstCall.args[1], sample.mocks.metadata);
+      assert(console.log.calledWith('Got metadata for subscription: %s', subscriptionName));
     });
-    it('should require subscriptionName', function () {
-      var sample = getSample();
-      var callback = sinon.stub();
-
-      sample.program.deleteSubscription(undefined, callback);
 
-      assert(callback.firstCall.args[0]);
-      assert(callback.firstCall.args[0].message = '"subscriptionName" is required!');
-      assert.equal(callback.firstCall.args[1], undefined);
-    });
     it('should handle error', function () {
       var sample = getSample();
       var error = new Error('error');
       var callback = sinon.stub();
-      sample.mocks.subscription.delete.callsArgWith(0, error);
+      sample.mocks.subscription.getMetadata.callsArgWith(0, error);
```

Comment:
Ah, gotcha - historically, my way of doing things is "I'm not going out of my way to fix things - but if I'm changing something that needs to be fixed eventually, why not fix it right now (since doing it now isn't *as* orthogonal as fixing something I wouldn't otherwise change)?"

That said, I'm also fine with waiting and doing this in a separate PR. :smile: